### ISSUE TITLE
Delayed report

### ISSUE THREAD

**Issue by GoogleCodeExporter** [useataaaaa/acra#11]
```
Is there a way just to add an exception to reports, but do not send report 
immediately?

```

Original issue reported on code.google.com by `kevin.gaudin` on 15 Nov 2010 at 10:26

**Comment by GoogleCodeExporter** [comment]
Original comment by `kevin.gaudin` on 18 Nov 2010 at 11:19
- Changed state: **New**

**Comment by GoogleCodeExporter** [comment]
```
This would be very useful for me as one of the selling points of my app is that 
it doesn't have to be connected to the Internet.
```

Original comment by `rupert.r...@gmail.com` on 3 Dec 2011 at 4:21

**Comment by GoogleCodeExporter** [comment]
```
@rupert.rawnsley: if you d'ont have the Internet permission, how do you want to 
get your reports back ?

For your app, you can use the mailTo option, which creates a report in an email 
(which then has to be sent by the user).
```

Original comment by `kevin.gaudin` on 4 Dec 2011 at 5:27

**Comment by GoogleCodeExporter** [comment]
```
@kevin.gaudin let me rephrase that: it doesn't have to *always* be connected to 
the Internet.

My app is a teaching aid for the classroom and the use case is this: teacher 
configures the software via a web portal, settings are deployed to the app via 
WiFi, teacher takes the tablets to a classroom that might not have WiFi, kids 
use the resource and crash the app (inevitably), teacher takes the tablet back 
into WiFi range, crash log is uploaded.

I would expect the upload to be initiated by the client app and not 
autonomously by ACRA. I do appreciate that this feature complicates ACRA, not 
least because it introduces "state" (the pending log files), and state always 
complicates things.

I can't use e-mail because the machines are not personalized, they are designed 
to be interchangeable, stateless units.
```

Original comment by `rupert.r...@gmail.com` on 4 Dec 2011 at 6:55

**Comment by GoogleCodeExporter** [comment]
```
So, the only missing feature is to let you trigger the reports uploading as 
ACRA already stores crash reports when no connection is available.

Stored ('pending' in ACRA terms) reports are sent 5 max at a time when one of 
these two events happen:
- Application is restarted (a full Application restart, not paused by the 
system when switching to another app)
- Another crash occurs
```

Original comment by `kevin.gaudin` on 4 Dec 2011 at 7:25

**Comment by GoogleCodeExporter** [comment]
```
In that case it already does everything I want because it is made out of pure 
awesome :-)
```

Original comment by `rupert.r...@gmail.com` on 4 Dec 2011 at 8:05

**Comment by GoogleCodeExporter** [comment]
```
It might be a good idea to mention this behavior in the Getting Started guide.

I was looking for something like:

"What happens when no network connection is active?"

I just happened to click on this Issue.
```

Original comment by `jonathan...@designdelegates.com` on 16 Dec 2011 at 9:29

**Comment by GoogleCodeExporter** [comment]
```
Please add a feature to let the programmer decide when to upload the error 
reports, instead of just doing it automatically the next time ACRA.init() is 
called.

The default operation could certainly be an automatic upload, but I'd like to 
be able to override it.
```

Original comment by `alexbiem...@gmail.com` on 9 Feb 2012 at 7:37

**Comment by GoogleCodeExporter** [comment]
```
To take it one step further, can we make it so ACRA only sends the crash report 
if the device has a WIFI connection? We do not want to send the crash report if 
the device has cellular connection.

Thanks
```

Original comment by `i...@kytephone.com` on 27 Jun 2012 at 9:25

**Comment by GoogleCodeExporter** [comment]
```
There was API to send pending reports in version 4.2.3 
(ErrorReporter.getInstance().checkReportsOnApplicationStart()), but it isn't 
public anymore (since 4.3.0).

It would be nice to have a way to send pending reports "manually" - as it was 
in versions before 4.3.0. 

Reason:

ACRA.init() is called only when application process is killed and restarted 
later. This happens only after FC, or when system decides to kill the process 
for some reason (this is a rare case). So when there is no Internet connection 
when application process restarts and ACRA.init() is called, the pending 
reports are never sent.
```

Original comment by `devj...@gmail.com` on 3 Oct 2012 at 6:26

**Comment by GoogleCodeExporter** [comment]
```
I think the spring cleaning went a bit too far here ;-)

I will absolutely put this method back to the public API.
```

Original comment by `kevin.gaudin` on 4 Oct 2012 at 4:50
- Changed state: **Accepted**
- Added labels: **Milestone-Release4.3.1**, **Type-Defect**
- Removed labels: **Type-Enhancement**

**Comment by GoogleCodeExporter** [comment]
Original comment by `kevin.gaudin` on 16 Oct 2012 at 8:24
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Delivered in v4.3.1: https://github.com/downloads/ACRA/acra/acra-4.3.1.zip
```

Original comment by `kevin.gaudin` on 16 Oct 2012 at 9:14
- Changed state: **Verified**
